File: airbyte-db/db-lib/src/main/java/io/airbyte/db/instance/jobs/migrations/V1_1_0_000__AddMetadataInJobs.java
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.db.instance.jobs.migrations;

import org.flywaydb.core.api.migration.BaseJavaMigration;
import org.flywaydb.core.api.migration.Context;
import org.jooq.DSLContext;
import org.jooq.Field;
import org.jooq.JSONB;
import org.jooq.Table;
import org.jooq.impl.DSL;
import org.jooq.impl.SQLDataType;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * Adds metadata column in jobs table.
 */
public class V1_1_0_000__AddMetadataInJobs extends BaseJavaMigration {

  private static final Logger LOGGER = LoggerFactory.getLogger(V1_1_0_000__AddMetadataInJobs.class);

  @Override
  public void migrate(final Context context) throws Exception {
    LOGGER.info("Running migration: {}", this.getClass().getSimpleName());

    final DSLContext ctx = DSL.using(context.getConnection());

    LOGGER.info("Add metadata column");
    addMetadataInJobs(ctx);

    LOGGER.info("Completed migration: {}", this.getClass().getSimpleName());
  }

  private static void addMetadataInJobs(final DSLContext ctx) {
    final Table<?> jobsTable = DSL.table("jobs");
    final Field<JSONB> metadata = DSL.field("metadata", SQLDataType.JSONB.nullable(true));

    ctx.alterTable(jobsTable)
        .addColumnIfNotExists(metadata)
        .execute();

  }

}

File: airbyte-db/db-lib/src/main/kotlin/io/airbyte/db/instance/jobs/migrations/V1_1_0_000__AddMetadataInJobs.kt
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.db.instance.jobs.migrations

import io.github.oshai.kotlinlogging.KotlinLogging
import org.flywaydb.core.api.migration.BaseJavaMigration
import org.flywaydb.core.api.migration.Context
import org.jooq.DSLContext
import org.jooq.Table
import org.jooq.impl.DSL
import org.jooq.impl.SQLDataType

private val log = KotlinLogging.logger {}

/**
 * Adds metadata column in jobs table.
 */
@Suppress("ktlint:standard:class-naming")
class V1_1_0_000__AddMetadataInJobs : BaseJavaMigration() {
  override fun migrate(context: Context) {
    log.info { "Running migration: ${javaClass.simpleName}" }
    val ctx = DSL.using(context.connection)

    log.info { "Add metadata column" }
    addMetadataInJobs(ctx)

    log.info { "Completed migration: ${javaClass.simpleName}" }
  }

  private fun addMetadataInJobs(ctx: DSLContext) {
    val jobsTable: Table<*> = DSL.table("jobs")
    val metadata = DSL.field("metadata", SQLDataType.JSONB.nullable(true))

    ctx
      .alterTable(jobsTable)
      .addColumnIfNotExists(metadata)
      .execute()
  }
}
